File: lld-in-c++/builder/Bird.cpp
#include "Bird.h"

Bird::Bird(string name, string gender, string color)
{
    this->name = name;
    this->gender = gender;
    this->color = color;
}

void Bird::showInfo(){
    cout<<"Name: "<<this->name<<", gender: "<<this->gender<<" and color: "<<this->color<<endl;
}

File: lld-in-java/Builder/src/Bird.java
public class Bird {
    String name, color, gender;
    Bird(Builder obj){
        this.name = obj.name;
        this.color = obj.color;
        this.gender = obj.gender;
    }
    public void showInfo(){
        System.out.println("Name: "+this.name+", Color: "+this.color+" and Gender: "+this.gender);
    }
    public static class Builder {
        String name, color, gender;
        public Builder setName(String n){
            this.name = n;
            return this;
        }
        public Builder setColor(String c){
            this.color = c;
            return this;
        }
        public Builder setGender(String g){
            this.gender = g;
            return this;
        }
        public Bird build(){
            Bird b = new Bird(this);
            return b;
        }

    }
}
